crates/ruff_linter/src/rules/flake8_pytest_style/rules/helpers.rs

```diff
@@ -58,15 +58,25 @@ pub(super) fn is_empty_or_null_string(expr: &Expr) -> bool {
         Expr::FString(ast::ExprFString { value, .. }) => {
             value.parts().all(|f_string_part| match f_string_part {
```

Comment:
I got rather confused by the `parts` method here. Like what's the difference between a `FStringPart` and an `FStringElement`, they even have the same members (although `FStringPart::Literal` takes a `StringLiteral` and `Element::Literal` an `FStringElementLiteral`). 

It clicked when opening the PR and looking up the definition, although I wonder if we can improve the naming (unrelated to this PR). 

A not fully formed idea:

* `ExprStringLiteral` -> `ExprString`
* `ExprStringLiteral::parts` rename to `literals` (it returns the individal string literals of which the string expression is composed)
* `ExprFString::parts` rename to `literals`
* `FStringPart` rename to `ExprFStringLiteral` 
* `FStringPart::Literal` rename to `FSTringPart::String`

Same for bytes. 

We may also want to change the `parts` and `parts_mut` methods to return a `&[]` instead of an iterator. 

The part I feel uncertain about is if it is okay to name an `f"abcd"` a literal, but I think it is. For example, the object notation `{a: "bcd"}` in JavaScript is a `ObjectLiteral` and it contains dynamic content as well. 

I don't suggest making these changes as part of this PR. It's just something I dripped over when reviewing and thought I would raise.


Edit: Even the Python spec names them literals: A formatted string literal

Reply:
Thanks! I'll explore this in a follow-up PR.